test/github/test_logic.py

```diff
@@ -634,6 +634,93 @@ def test_maybe_rerun_stale_checks_multiple_base_refs(
+        pull_request = build(
+            builder.pull_request()
+            .base_ref_name("main")
+            .commit(
+                builder.commit()
+                .status(Commit.BUILD_SUCCESSFUL)
+                .check_suites([builder.check_suite().check_runs([check_run])])
+            )
+            .review(
+                builder.review()
+                .submitted_at("2020-01-13T14:59:58Z")
+                .state(ReviewState.APPROVED)
+            )
+            .merged(False)
+        )
+        self.assertTrue(
+            github_logic.maybe_rerun_stale_checks_on_approved_pull_request(pull_request)
+        )
+        mock_rerequest_check_run.assert_called_once_with(
+            pull_request.repository_owner_handle(),
+            pull_request.repository_name(),
+            check_run.database_id(),
+        )
+
+    @patch("src.github.logic.SGTM_FEATURE__ALLOW_CHECK_RERUN_ON_APPROVAL", True)
+    @patch("src.github.logic.SGTM_FEATURE__CHECK_RERUN_THRESHOLD_HOURS", 1)
+    @patch(
+        "src.github.logic.SGTM_FEATURE__CHECK_RERUN_BASE_REF_NAMES",
+        ["main"],
+    )
+    def test_no_rerun_stale_checks_on_unapproved_pull_request(
+        self, mock_rerequest_check_run
+    ):
+        check_run = build(builder.check_run().completed_at("2020-01-13T14:59:58Z"))
+        pull_request = build(
+            builder.pull_request()
+            .base_ref_name("main")
+            .commit(
+                builder.commit()
+                .status(Commit.BUILD_SUCCESSFUL)
+                .check_suites([builder.check_suite().check_runs([check_run])])
+            )
+            .review(
+                builder.review()
+                .submitted_at("2020-01-13T14:59:58Z")
+                .state(ReviewState.CHANGES_REQUESTED)
+            )
+            .merged(False)
+        )
+        self.assertFalse(
+            github_logic.maybe_rerun_stale_checks_on_approved_pull_request(pull_request)
+        )
+        mock_rerequest_check_run.assert_not_called()
+
+    @patch("src.github.logic.SGTM_FEATURE__ALLOW_CHECK_RERUN_ON_APPROVAL", False)
+    def test_noop_if_feature_not_enabled(self, mock_rerequest_check_run):
+        check_run = build(builder.check_run().completed_at("2020-01-13T14:59:58Z"))
+        pull_request = build(
+            builder.pull_request()
+            .base_ref_name("main")
+            .commit(
+                builder.commit()
+                .status(Commit.BUILD_SUCCESSFUL)
+                .check_suites([builder.check_suite().check_runs([check_run])])
+            )
+            .review(
+                builder.review()
+                .submitted_at("2020-01-13T14:59:58Z")
+                .state(ReviewState.APPROVED)
+            )
+            .merged(False)
+        )
+        self.assertFalse(
+            github_logic.maybe_rerun_stale_checks_on_approved_pull_request(pull_request)
+        )
+        mock_rerequest_check_run.assert_not_called()
+
```

Comment:
Can you add another test case for the intersection of logic between this feature and someone that has the `merge after ...` labels? I think this logic is straight forward but its possible that there are race conditions when these 2 rerun features intersect.